msal/client_application.go

```diff
@@ -56,12 +57,12 @@ func (client *clientApplication) acquireTokenSilent(ctx context.Context, silent
 	silent.augmentAuthenticationParameters(&authParams)
 
 	// TODO(jdoak): Think about removing this after refactor.
-	if sm, ok := client.cache.(*storage.Manager); ok {
-		client.cacheAccessor.BeforeCacheAccess(sm)
-		defer client.cacheAccessor.AfterCacheAccess(sm)
+	if sm, ok := client.manager.(*storage.Manager); ok {
```

Comment:
Is there a reason why we don't want to call these on the fake?

Reply:
Skipping the first poinrt and straight to your second point, now that we have Unmarshaller/Marshaller, we could make this more grandular.  Introduced a Serializer interface, it now tests for that and if it is implemented it uses it.